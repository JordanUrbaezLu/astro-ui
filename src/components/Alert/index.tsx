/* istanbul ignore file */
export * from "./Alert";
