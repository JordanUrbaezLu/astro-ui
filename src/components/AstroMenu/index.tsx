export * from "./AstroMenu";
export * from "./AstroMenuItem";
