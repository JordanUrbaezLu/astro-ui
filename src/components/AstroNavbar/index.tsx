export * from "./AstroNavbar";
export * from "./AstroNavbarItem";
