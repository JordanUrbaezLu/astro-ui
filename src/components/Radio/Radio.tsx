import * as React from "react";
import classNames from "classnames";
import styles from "./Radio.module.scss";
import {
  MdRadioButtonUnchecked,
  MdRadioButtonChecked,
} from "react-icons/md";
import { IconButton } from "../IconButton";
import { Typography } from "../Typography/Typography";

export interface RadioProps
  extends React.ComponentPropsWithoutRef<"button"> {
  /**
   * The label for the Radio
   */
  children: React.ReactNode;
  /**
   * If the Radio is disabled
   *
   * @default false
   */
  disabled?: boolean;
  /**
   * If the Radio is selected
   *
   * @default false
   */
  isSelected?: boolean;
  /**
   * The callback fired when requesting to change the Radio state
   */
  onChange: () => void;
}

/**
 * @public
 *
 * @description
 *
 * The Radio allows a user to select an item.
 */
export const Radio: React.FC<RadioProps> = (props) => {
  const {
    children,
    className,
    disabled = false,
    isSelected = false,
    onChange,
    ...rest
  } = props;

  return (
    <div className={styles.container}>
      <IconButton
        className={classNames(
          className,
          isSelected && styles.isSelected
        )}
        content={
          isSelected ? (
            <MdRadioButtonChecked />
          ) : (
            <MdRadioButtonUnchecked />
          )
        }
        disabled={disabled}
        onClick={onChange}
        role="radio"
        variant="secondary"
        {...rest}
      />
      <Typography
        className={classNames(
          disabled && styles.disabled,
          isSelected && styles.selected
        )}
        variant="secondary"
      >
        {children}
      </Typography>
    </div>
  );
};

Radio.displayName = "Radio";
