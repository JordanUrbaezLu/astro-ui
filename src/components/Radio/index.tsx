/* istanbul ignore file */
export * from "./Radio";
