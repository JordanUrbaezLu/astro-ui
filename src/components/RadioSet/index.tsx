/* istanbul ignore file */
export * from "./RadioSet";
