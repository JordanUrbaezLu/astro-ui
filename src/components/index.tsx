export * from "./Alert";
export * from "./Button";
export * from "./Checkbox";
export * from "./DatePicker";
export * from "./Link";
export * from "./Menu";
export * from "./Tooltip";
export * from "./Select";
export * from "./Switch";
export * from "./TextField";
