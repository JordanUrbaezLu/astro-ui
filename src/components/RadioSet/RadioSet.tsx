import classNames from "classnames";
import * as React from "react";
import styles from "./RadioSet.module.scss";
import { Radio } from "../Radio/Radio";

export interface RadioSetProps
  extends React.ComponentPropsWithoutRef<"ul"> {
  /**
   * The Radios in the Radios Set
   */
  children: React.ReactNode;
}

/**
 * @public
 *
 * @description
 *
 * The Radio Set is a set of options, as Radios, for the user.
 */
export const RadioSet: React.FC<RadioSetProps> = (props) => {
  const { children, className, ...rest } = props;

  return (
    <ul
      className={classNames(styles.radioSet, className)}
      role="list"
      {...rest}
    >
      {React.Children.map(children, (radio, index) => {
        if ((radio as React.ReactElement).type === Radio) {
          return (
            <li
              className={styles.radioSetItem}
              key={index}
              role="listitem"
            >
              {radio}
            </li>
          );
        } else {
          return null;
        }
      })}
    </ul>
  );
};

RadioSet.displayName = "RadioSet";
