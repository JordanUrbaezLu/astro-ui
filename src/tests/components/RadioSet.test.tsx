import { render, screen } from "@testing-library/react";
import { Radio } from "../../components/Radio";
import { RadioSet } from "../../components/RadioSet";
import { axe } from "jest-axe";

describe("RadioSet", () => {
  test("Should render component correctly.", () => {
    const { container } = render(
      <RadioSet>
        <Radio onChange={() => undefined}>Radio</Radio>
      </RadioSet>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("Accessibility", () => {
  test("Should have no accessibility violations.", async () => {
    const { container } = render(
      <RadioSet>
        <Radio onChange={() => undefined}>Radio</Radio>
      </RadioSet>
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

describe("Props", () => {
  test("Should render children correctly.", () => {
    render(
      <RadioSet>
        <Radio onChange={() => undefined}>Radio</Radio>
        <div>Div</div>
      </RadioSet>
    );
    expect(screen.getByRole("list")).not.toHaveTextContent("Div");
    expect(screen.getByRole("list")).toHaveTextContent("Radio");
  });
});
