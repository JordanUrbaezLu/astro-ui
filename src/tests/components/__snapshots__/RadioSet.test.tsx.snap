// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioSet Should render component correctly. 1`] = `
<ul
  class="radioSet"
  role="list"
>
  <li
    class="radioSetItem"
    role="listitem"
  >
    <div
      class="container"
    >
      <button
        aria-label="Icon Button"
        class="iconButton medium secondary"
        role="radio"
      >
        <div
          class="container"
          role="presentation"
        />
        <svg
          fill="currentColor"
          height="22"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </button>
      <div
        class="typography typographyh2 secondary normal"
      >
        Radio
      </div>
    </div>
  </li>
</ul>
`;
